### ISSUE TITLE
Grand Theft Auto IV crashing for every 5-10 minutes

### ISSUE THREAD

**Issue by Kruger-28** [doitsujin/dxvk#4494]
### Software information
Grand Theft Auto IV, Medium-High Settings preset. 1080p Resolution. (32-bit)

### System information
- GPU: Intel Iris Xe Graphics (8GB VRAM)
- Driver: 32.0.101.6299
- Wine version: N/A
- DXVK version: 2.2

###My take on the problem
The game is opening through Vulkan API as my driver and graphic device can support the Vulkan version. The DIrectX9 version is still very crappy. It runs around 20fps in 900p resolution and still has some graphic issues (ex. vehicle disfiguration on crashing). The game when opening through Vulkan runs smoothly, but within 5-10 minutes, it crashes like hell, that too when we are supposed to do a story mission. I have used the latest version of DXVK, and I couldn't figure out why exactly the game was crashing. The settings preset which I have used rough takes around 412MB of VRAM, keeping Shadow Quality in Medium (the shadows are way too graphic intensive). I just started the game without DXVK, and the game didn't crash. Although the quality and the performance of the game is compromised at an extent but, I think the DXVK community needs to look on this. It's such a nice game and very well-known for the graphics and physics, it hurts me to play the game with so many issues (especially when I'm very close to complete a story mission). The same is happening with the latest version v.2.5.1.

Here is the log file of v2.2.
[GTAIV_d3d9.log](https://github.com/user-attachments/files/17934746/GTAIV_d3d9.log)
[GTAIV_dxgi.log](https://github.com/user-attachments/files/17934747/GTAIV_dxgi.log)

Thank you.

**Comment by Blisto91** [comment]
Is the log from a time it crashed? There isn't any interesting output in it.

I'll note that GTA IV is probably one of the most tested games with dxvk in general and we are not aware of any general crash issue. The only reports we've had recently are from Intel Windows users like youself or random Windows setup shenanigans. Also note that Windows is not officially a supported platform.
None of us have a setup with both Intel and Windows to test with so i suspect it would likely be better for you to make a issue here instead https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT

**Comment by Blisto91** [comment]
There is nothing to suggest when we don't know why it crashes. The best i can do is to link you to the[ Third-party software section in the Windows wiki](https://github.com/doitsujin/dxvk/wiki/Windows#third-party-software). 
As said i think it would better to report it via the link above.

**Comment by Kruger-28** [comment]
Trying it.

**Comment by Freerubuk** [comment]
i have same issue, maybe the card doesnt DirectX 9 game properly

**Comment by Kruger-28** [comment]
Yeah, it's a good one. But it stutters a lot while gameplay, maybe when loading textures and fixing light or compiling shaders. But it does crash on cutscenes.

**Comment by Freerubuk** [comment]
Oh it crash after 1 hours but im not get stuffers

**Comment by Blisto91** [comment]
The mod linked is just dxvk 2.4 and some game launch parameters fwiw.

**Comment by Kruger-28** [comment]
@Freerubuk Please tell me what else do you do for running the game smooth.

**Comment by Freerubuk** [comment]
Honestly i don’t know any im quit gta 4 because of that

**Comment by Blisto91** [comment]
It will likely get fixed faster if someone reported it to Intel https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT

**Comment by Freerubuk** [comment]
Author of this post report it yesterday tho

**Comment by Kruger-28** [comment]
Hey guys. I have a doubt. Does every .dll from x32 folder should be added to the game directory or just the one which the DirectX API calls for?... I have added all the .dll files from the latest DXVK, and it works fine. No stutters, smooth gameplay. But it does crash after an hour of gameplay.

**Comment by Blisto91** [comment]
Driver is crashing or timing out
```
err:   DxvkDevice: waitForIdle: Operation failed
err:   DxvkSubmissionQueue: Command submission failed: VK_ERROR_DEVICE_LOST
```

**Comment by Kruger-28** [comment]
Is there anything to resolve it?

**Comment by Blisto91** [comment]
No because we don't know what the problem is. Best to wait and see what Intel says

**Comment by Freerubuk** [comment]
Does dxwrapper work guys? Im lazy to test

**Comment by mirh** [comment]
Dxwrapper has nothing to do with dxvk.
And until https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/issues/872 gets handled, you can try to lower the settings for chances you may get a longer playrun.

**Comment by turboZd3v1l** [comment]
The game doesn't crash when using native API. When I use dxwrapper, the game doesn't even open. I have tried even other wrappers. Nothing seemed to work at all.

**Comment by Freerubuk** [comment]
I wish gta 4 have good performance and get remastered

**Comment by Kruger-28** [comment]
When can we expect a GTA IV remaster?

**Comment by mirh** [comment]
There are a thousand and one guides about getting gta 4 to go faster (even though the biggest cheat is probably just to use a windows older than 1607~1703). And then you can try d912pxy, dgvoodoo and even wined3d I guess.

But for the love of god, stick this to dxvk and nothing else here?
As said above, try to see if you can correlate any setting or thing to the crasbes (`DXVK_HUD=full` may also help with clues).
Otherwise just wait for intel to find dev time for it.

**Comment by turboZd3v1l** [comment]
Guys. My problem is fixed. I used Magpie FSR with DXVK and it doesn't crash. I feel very happy that in so many years,it's the first time I've played GTA IV without stuttering, without crashing. I am just loving it. Thank you guys for your input. Much appreciated.

**Comment by Freerubuk** [comment]
Thanks you for answer i must try it now!

**Comment by Freerubuk** [comment]
Finally time to install gta 4 with fusionfix

**Comment by Freerubuk** [comment]
Hey @turboZd3v1l what your setting in magpie

**Comment by Freerubuk** [comment]
Bro how to use this

**Comment by mirh** [comment]
This is probably the first time in my life where I feel like "restrict comments" would make sense..

> I used Magpie FSR with DXVK and it doesn't crash.

If just so it wasn't that this is very odd. Is it a matter of having \*very much\* lowered the resolution, or actually playing it in windowed mode?

**Comment by turboZd3v1l** [comment]
@mirh Bruh... I just wondered if FSR mode will work or not. Although only some gamers use Magpie and it eventually works with every D3D9 game or even older on the newer PC.

**Comment by Freerubuk** [comment]
How to use this program @turboZd3v1l

**Comment by turboZd3v1l** [comment]
@Freerubuk Just download Magpie v0.11.1 64 bit

**Comment by Kruger-28** [comment]
https://github.com/JamesCJ60/AMD-APU-Tuning-Utility/releases/tag/4.0.3

Use this link and download it.

**Comment by Freerubuk** [comment]
@turboZd3v1l add gta 4 then keep it default then run gta 4?

**Comment by Freerubuk** [comment]
Oh

**Comment by Freerubuk** [comment]
Maybe i try this Tommrow im gonna sleep bye

**Comment by Blisto91** [comment]
This isn't a chatroom guys. Please keep the offtopic to a minimum

**Comment by Freerubuk** [comment]
Ok i try thiz Tommrow

**Comment by mirh** [comment]
I really feel like we crossed the line of pointlessness and this can be closed.

**Comment by Freerubuk** [comment]
Maybe

**Comment by SavageWaffle** [comment]
Fix TBD can we merge the other ticket for it
